src/main/java/xyz/gvital/ListNode.java

```diff
@@ -0,0 +1,33 @@
+package xyz.gvital;
+
+public class ListNode {
+    protected int val;
```

Comment:
Could be private. Using protected make no sense since we don't have subclasses
```suggestion
    private int val;
```